tests/test_cssselect.py

```diff
@@ -491,7 +502,11 @@ def xpath(css):
         assert xpath('e:not(:nth-child(odd))') == (
             "e[not(count(preceding-sibling::*) mod 2 = 0)]")
         assert xpath('e:nOT(*)') == (
-            "e[0]")  # never matches
+            "e[0]")  # never matches        
+        assert xpath('e:has(> f)') == 'e[./f]'
+        assert xpath('e:has(f)') == 'e/descendant-or-self::f/ancestor-or-self::e'
```

Comment:
Can't this match some descendant instead of the element itself or am I misunderstanding something? I would expect all `e:has()` selectors to be translated to `e[]` as it should always return the element itself.

Reply:
Maybe it's me misunderstanding something :) 
I guess it won't match any descendants, judging by this snippet
``` bash
>>> from parsel import Selector
>>> s = Selector("<div><e><f>1</f><b><f>2</f></b></e></div>")
>>> s.xpath(".//e/descendant-or-self::f/ancestor-or-self::e")
[<Selector xpath='.//e/descendant-or-self::f/ancestor-or-self::e' data='<e><f>1</f><b><f>2</f></b></e>'>]
```
If descendants could be matched, there would be a couple of `<f></f>`'s in the resulting list

Reply:
I tried to provide an example of `"<div><e><e><b><f></f></b></e></e></div>"` but found that it returns both `e` elements and this is a correct behavior so maybe there is indeed no problem :) It just looks strange to me to descend the tree and then ascend back, and isn't it correct and more readable to use `e[descendant::f]` too (and something simllar for `e:has(~ f)`)? Maybe I'm missing some cases where it's different or wrong.